apps/meteor/app/livechat/server/hooks/sendToCRM.ts

```diff
@@ -38,55 +76,48 @@ const getAdditionalFieldsByType = (type, room) => {
 			return {};
 	}
 };
-async function sendToCRM(type, room, includeMessages = true) {
+
+async function sendToCRM(
+	type: string,
+	room: OmnichannelRoomWithExtraFields,
+	includeMessages: boolean | IOmnichannelSystemMessage[] = true,
+): Promise<OmnichannelRoomWithExtraFields> {
 	if (!settings.get('Livechat_webhookUrl')) {
 		return room;
 	}
 
-	const postData = await Livechat.getLivechatRoomGuestInfo(room);
-
-	postData.type = type;
+	const postData: Awaited<ReturnType<typeof Livechat.getLivechatRoomGuestInfo>> & { type: string; messages: IOmnichannelSystemMessage[] } =
```

Comment:
why such TS witchcraft? 🙈 I'm sorry I'm dumb

Reply:
:kek:

It's saying: "Give me the same return type of `livechat.getlivechatroomGuestInfo` after being awaited, and add { type, messages } props to it (as they don't come from the function call)"

Reply:
once we have it typed this will be removed then, right?
![image](https://user-images.githubusercontent.com/8591547/229197722-f793efed-eaea-4a02-b2e0-a2ce3d895e4a.png)

Reply:
![image](https://user-images.githubusercontent.com/11577696/229243818-9986c6d2-e743-4e35-bda2-6056e7f8af1f.png)